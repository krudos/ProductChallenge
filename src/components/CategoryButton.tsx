import React, {FC} from 'react';
import {Container, Title} from './CategoryButton.styled';

interface CategoryButtonProps {
  title: string;
  id: string;
  selected: boolean;
  onPress?: () => void;
}
const CategoryButton: FC<CategoryButtonProps> = ({
  title,
  selected,
  onPress,
}) => {
  return (
    <Container selected={selected} onPress={onPress}>
      <Title selected={selected}>{title}</Title>
    </Container>
  );
};

export default CategoryButton;
