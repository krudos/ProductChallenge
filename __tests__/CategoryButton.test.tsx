import React from 'react';

import renderer from 'react-test-renderer';
import CategoryButton from '../src/components/CategoryButton';

describe('CategoryButton', () => {
  it('should render selected', () => {
    const render = renderer
      .create(<CategoryButton title="test" selected={true} id={'1'} />)
      .toJSON();

    expect(render).toMatchSnapshot();
  });

  it('should render selected false ', () => {
    const render = renderer
      .create(<CategoryButton title="test" selected={false} id={'1'} />)
      .toJSON();

    expect(render).toMatchSnapshot();
  });
});
