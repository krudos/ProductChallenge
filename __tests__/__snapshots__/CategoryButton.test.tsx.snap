// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton should render selected 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "marginRight": 8,
      "opacity": 1,
    }
  }
>
  <Text
    selected={true}
    style={
      Array [
        Object {
          "color": "",
          "marginLeft": 8,
          "marginRight": 8,
          "textAlign": "center",
          "textTransform": "capitalize",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`CategoryButton should render selected false  1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "marginRight": 8,
      "opacity": 1,
    }
  }
>
  <Text
    selected={false}
    style={
      Array [
        Object {
          "color": "",
          "marginLeft": 8,
          "marginRight": 8,
          "textAlign": "center",
          "textTransform": "capitalize",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
